app/src/main/java/com/example/lastfm/ServiceGenerator.java

```diff
@@ -0,0 +1,60 @@
+package com.example.lastfm;
+
+import android.util.Log;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+import java.util.List;
+
+import okhttp3.OkHttpClient;
+import okhttp3.logging.HttpLoggingInterceptor;
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public class ServiceGenerator {
+
+    private static final String TAG = "Retrofit";
+
+    private static final String BASE_URL = "http://ws.audioscrobbler.com/2.0/";
```

Comment:
u use substring "2.0" here and in the method of API. Check one more time if it's correct